packages/client/utils/mediaRoom/urlFactory.ts

```diff
@@ -0,0 +1,23 @@
+const signalingServerHostname = window.location.hostname
+
+interface getSignalingServerUrlSignature {
+  roomId: string
+  peerId: string
+  authToken: string
+  teamId: string
+}
+
+export function getSignalingServerUrl({
+  roomId,
+  peerId,
+  authToken,
+  teamId
+}: getSignalingServerUrlSignature) {
+  const url = new URL(`wss://${signalingServerHostname}`)
+  url.port = String(process.env.PROTOO_LISTEN_PORT)
+  url.searchParams.append('roomId', roomId)
+  url.searchParams.append('peerId', peerId)
+  url.searchParams.append('authToken', authToken)
+  url.searchParams.append('teamId', teamId)
+  return url.toString()
```

Comment:
this returns a string, but the interface is an object
not sure what is calling this function. still tbd?

Reply:
Ah the interface is an object, but it's an interface for the fn signature not the return value. I didn't specify typing for the return value, which I should probably do. This fn is called by `MediaRoom` right before it creates a web socket connection to the media server.